### PR TITLE
Implement Banned characters component (#6)

### LINKED ISSUE #6: Banned characters
screen shot 2017-09-20 at 7 21 08 pm

### PR DESCRIPTION
**Pull request by KantinDmitry**
Closes #6

### CODE REVIEW

**Comment by R1ZZU** [comment]
Correct me if I'm wrong, but I thought we had also `scale` transition for banned chars?

File: src/components/BannedCharacters/BannedCharacters.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { h } from 'preact';
+import classNames from 'classnames';
+
+const FONT_SIZE = 18;
+
+const BannedCharacters = ({ bannedCharacters = [], highlightedCharacters = [] }) => (
+    <div class="banned-characters">
+        <span class="label">Banned chars </span>
```
Comment (R1ZZU) [inline src/components/BannedCharacters/BannedCharacters.js]: `:`

File: src/components/BannedCharacters/BannedCharacters.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { h } from 'preact';
+import classNames from 'classnames';
+
+const FONT_SIZE = 18;
+
+const BannedCharacters = ({ bannedCharacters = [], highlightedCharacters = [] }) => (
+    <div class="banned-characters">
```
Comment (R1ZZU) [inline src/components/BannedCharacters/BannedCharacters.js]: I'd also check `bannedCharacters.length` and if `0` – won't render component at all

File: src/components/BannedCharacters/BannedCharacters.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                color: #b3decc;
+                display: inline-block
+            }
+
+            .banned-character {
+                display: inline-block;
+                font-family: monospace;
+                font-weight: 600;
+                font-size: ${FONT_SIZE}px;
+                color: #FFFFFF;
```
Comment (R1ZZU) [inline src/components/BannedCharacters/BannedCharacters.js]: Maybe it makes sense to move colors to constants alongside with `FONT_SIZE`?

Comment (R1ZZU) [reply src/components/BannedCharacters/BannedCharacters.js]: You can checkout #18 for reference

File: src/components/BannedCharacters/BannedCharacters.js (hunk 2 above)
Comment (KantinDmitry) [reply src/components/BannedCharacters/BannedCharacters.js]: Sure

File: src/components/BannedCharacters/BannedCharacters.js (hunk 3 above)
Comment (KantinDmitry) [reply src/components/BannedCharacters/BannedCharacters.js]: Updated. But I have a question about file naming. Could you explain cases of usage camel case vs dashes?

**Comment by R1ZZU** [comment]
> I have a question about file naming. Could you explain cases of usage camel case vs dashes?

I was thinking about camelCase naming only in case of react components. all other files are train-case

File: src/components/BannedCharacters/BannedCharacters.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    text-align: center;
+                    display: flex;
+                    align-items: center;
+                    justify-content: center;
+                }
+
+                .highlighted {
+                    background-color: ${Colors.HIGHLIGHTED_CHARACTER_BACKGROUND};
+                    border: 2px solid ${Colors.HIGHLIGHTED_CHARACTER_BORDER};
+                    transform: scale(${DisplayConstants.HIGHLIGHTED_CHAR_SCALE_COEFFICIENT});
```
Comment (kirilknysh) [inline src/components/BannedCharacters/BannedCharacters.js]: Shouldn't `transform` be animated?

Comment (kirilknysh) [reply src/components/BannedCharacters/BannedCharacters.js]: Also, pay attention: during the scaling the height of component is changed which makes the sibling markup jump.

File: src/components/BannedCharacters/BannedCharacters.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { h } from 'preact';
+import classNames from 'classnames';
+import { DisplayConstants, Colors } from './banned-characters-style-constants';
+
+const BannedCharacters = ({ bannedCharacters = [], highlightedCharacters = [] }) => {
+    if (bannedCharacters.length) {
```
Comment (Shastel) [inline src/components/BannedCharacters/BannedCharacters.js]: maybe in this case early return is better?

Comment (KantinDmitry) [reply src/components/BannedCharacters/BannedCharacters.js]: make sense for readability

File: src/components/BannedCharacters/BannedCharacters.js (hunk 4 above)
Comment (KantinDmitry) [reply src/components/BannedCharacters/BannedCharacters.js]: Sure. Didn't get it from the screenshots )

**Pull request merged**